src/software/ai/navigator/path_planner/trajectory_path_node.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include "software/ai/navigator/path_planner/bang_bang_trajectory_2d.h"
+
+class TrajectoryPathNode
+{
+   public:
+    /**
+     * Constructor for a trajectory and its end time
+     * @param trajectory Trajectory of this trajectory path node
+     * @param trajectory_end_time End time of this trajectory
+     */
+    TrajectoryPathNode(const BangBangTrajectory2D &trajectory, double trajectory_end_time)
+        : trajectory(trajectory), trajectory_end_time_sec(trajectory_end_time){};
+
+    /**
+     * Constructor for a trajectory. It is assumed that the end time
+     * is the total time of the trajectory
+     * @param trajectory Trajectory of this trajectory path node
+     */
+    TrajectoryPathNode(const BangBangTrajectory2D &trajectory)
```

Comment:
It seems like `TrajectoryPathNode` acts as a wrapper around a `BangBangTrajectory2D`, could each trajectory instead include the end time? Also could you clarify the difference between total run time and end time?

Reply:
Yeah that's an alternative possibility that would also be valid. I just decided to make it wrapper, and not have the class handle extra specific behaviour around early end time.